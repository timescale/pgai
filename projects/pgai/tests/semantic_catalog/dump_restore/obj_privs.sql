select
  x.objtype
, x.objnames[1] as "schema"
, x.objnames[2] as "object"
, case
    when x.objtype in ('table', 'view') then
        has_table_privilege('bot', x.objid, 'select')
    else
        has_function_privilege('bot', x.objid, 'execute')
  end as has_access
from ai.semantic_catalog_obj_1 x
where x.objsubid = 0
order by x.objtype, x.objnames
;