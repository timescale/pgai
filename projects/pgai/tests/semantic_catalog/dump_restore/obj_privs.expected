  objtype  |    schema    |      object      | has_access 
-----------+--------------+------------------+------------
 procedure | postgres_air | advance_air_time | t
 table     | postgres_air | airport          | t
 table     | postgres_air | bob              | t
(3 rows)

